airflow/jobs/triggerer_job.py

```diff
@@ -66,6 +67,15 @@ def register_signals(self) -> None:
         signal.signal(signal.SIGINT, self._exit_gracefully)
         signal.signal(signal.SIGTERM, self._exit_gracefully)
 
+    @provide_session
+    def is_needed(self, session) -> bool:
+        """
+        Tests if the triggerer job needs to be run (i.e., if there are triggers
+        in the trigger table).
+        This is used for the warning boxes in the UI.
+        """
+        return session.query(Trigger).count() > 0
```

Comment:
```suggestion
        return session.query(func.count(Trigger.id)).scalar() > 0
```

Reply:
Right - I remember seeing this somewhere else, it makes SQLAlchemy emit an actually-efficient query?

Reply:
yeah